src/p2pool.cpp

```diff
@@ -391,6 +391,31 @@ void p2pool::handle_chain_main(ChainMain& data, const char* extra)
 				LOGINFO(0, log::LightCyan() << "You received a payout of " << log::LightGreen() << log::XMRAmount(payout) << log::LightCyan() << " in block " << log::LightGreen() << data.height);
 			}
 			api_update_block_found(&data);
+
+			if(!params().m_onBlockFound.empty())
+			{
+			    std::stringstream cmd;
```

Comment:
std::stringstream allocates memory and is overall less efficient, use
```
char buf[log::Stream::BUF_SIZE + 1];
log::Stream cmd(buf);
cmd << all your stuff << '\0';
```

Reply:
adjusted